src/EditorFeatures/Core/Implementation/Formatting/IEditorFormattingService.cs

```diff
@@ -29,6 +30,12 @@ internal interface IEditorFormattingService : ILanguageService
         /// </summary>
         Task<IList<TextChange>> GetFormattingChangesAsync(Document document, TextSpan? textSpan, CancellationToken cancellationToken);
 
+        /// <summary>
+        /// Returns the text changes necessary to format the document with custom document options.
+        /// If "textSpan" is provided, only the text changes necessary to format that span are needed.
+        /// </summary>
+        Task<IList<TextChange>> GetFormattingChangesAsync(Document document, TextSpan? textSpan, DocumentOptionSet documentOptions, CancellationToken cancellationToken);
```

Comment:
TS + F# both use this interface so I had to add new methods instead of modifying the existing ones.

Reply:
Do they not use the ExternalAccess patterns?

Reply:
I believe they do for some interfaces, but for this one it [looks like](http://index/?leftProject=Microsoft.CodeAnalysis.EditorFeatures&leftSymbol=wqijbzzaayub&rightProject=Microsoft.CodeAnalysis.TypeScript.EditorFeatures&file=Features%5CFormatting%5CTypeScriptEditorFormattingService.cs&line=18) they just implement it directly?

Reply:
I don't understand why this is better than adding a parameter? Isn't it a breaking change either way?

Reply:
If we add the methods and merge the TypeScript change first, I don't think it would be a breaking change, since TypeScript will just have unused members that aren't associated with any interface method. When we then merge the interface changes in from the Roslyn side, those unassociated members will be tied to interface members.

However, if others feel strongly that editing the existing interface methods is the correct way to go here, then maybe a breaking change would be OK? It would just have to be a coordinated insertion with TypeScript (I think).

Reply:
> I don't think it would be a breaking change, since TypeScript will just have unused members that aren't associated with any interface method. When we then merge the interface changes in from the Roslyn side, those unassociated members will be tied to interface members.

This is still a breaking change AFAIK, since it's at the moment TypeScript is compiled that the compiler is sorting out what is and isn't a interface implementation. TypeScript needs to be recompiled no matter what.

Reply:
Gotcha, in that case I modified the code to just update the existing signature.